### PR TITLE
Fix wrong nested array in maxvalue

### PR DESCRIPTION
**Pull request by matrixino**
This fixes a problem wth maxprice var.
Previously, when $minPrice was > $maxPrice[$idCountry][$currency['id_currency']], it would add the whole minprice array to maxprice instead of the single value.
This leads to a wrong nested array and make products impossible to save in the BO (error 500 for unsupported operand types at line 548).
We only need to assign the single value.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/ps_facetedsearch/126)

### CODE REVIEW

File: ps_facetedsearch.php [hunk 1]
```diff
@@ -529,7 +529,7 @@ public function indexProductPrices($idProduct, $smart = true)
                         }
 
                         if ($minPrice > $maxPrice[$idCountry][$currency['id_currency']]) {
```
Comment (PierreRambaud) [inline ps_facetedsearch.php]:
```suggestion
                        if ($minPrice[$idCountry][$currency['id_currency']] > $maxPrice[$idCountry][$currency['id_currency']]) {
```

File: ps_facetedsearch.php [hunk 2]
```diff
@@ -529,7 +529,7 @@ public function indexProductPrices($idProduct, $smart = true)
...
-                            $maxPrice[$idCountry][$currency['id_currency']] = $minPrice;
+                            $maxPrice[$idCountry][$currency['id_currency']] = $minPrice[$idCountry][$currency['id_currency']];
```
Comment (PierreRambaud) [inline ps_facetedsearch.php]:
```suggestion
                            $maxPrice[$idCountry][$currency['id_currency']] = $price;
```

Comment (PierreRambaud) [reply ps_facetedsearch.php]: Should be this?

**Comment by PierreRambaud** [comment]
Any news? We planned to release a new version today and we want to integrate the fix.

**Comment by matrixino** [comment]
> Any news? We planned to release a new version today and we want to integrate the fix.

Yes, changing it with $price is even better. Mine was a quick fix.
532: $maxPrice[$idCountry][$currency['id_currency']] = $price;

**Comment by PierreRambaud** [comment]
Could you update your pull request or should I create a new one? (I'm not able to update your branch)

**Comment by matrixino** [comment]
> Could you update your pull request or should I create a new one? (I'm not able to update your branch)

Is it ok now?

**Comment by PierreRambaud** [comment]
Not yet. One more thing needs to be update. :)

**Comment by matrixino** [comment]
> Not yet. One more thing needs to be update. :)

By me? Which one? I updated the diff already.

**Comment by PierreRambaud** [comment]
This line: https://github.com/PrestaShop/ps_facetedsearch/pull/126/files#diff-5148302708676f17872e622586c6ffd5R531

**Comment by matrixino** [comment]
> This line: https://github.com/PrestaShop/ps_facetedsearch/pull/126/files#diff-5148302708676f17872e622586c6ffd5R531

Oops, done.

**Comment by PierreRambaud** [comment]
Thanks, wait for CI and merge :)

**Pull request merged**